src/repr/src/adt/datetime.rs

```diff
@@ -201,49 +202,21 @@ impl DateTimeField {
             .unwrap_or_else(|| panic!("Cannot get larger DateTimeField than {}", self))
     }
 
-    /// Returns the number of seconds in a single unit of `field`.
-    ///
-    /// # Panics
-    ///
-    /// Panics if called on a non-duration field or a field smaller than a second.
-    pub fn seconds_multiplier(self) -> i64 {
-        use DateTimeField::*;
-        match self {
-            Day => 60 * 60 * 24,
-            Hour => 60 * 60,
-            Minute => 60,
-            Second => 1,
-            _other => unreachable!(
-                "Do not call with a non-duration field or a field smaller than a second"
-            ),
-        }
-    }
-
-    /// Returns the number of 'fields' in a single second.
+    /// Returns the number of microseconds in a single unit of `field`.
     ///
     /// # Panics
     ///
-    /// Panics if called on a non-duration field or a field larger than or equal to a second.
-    pub fn seconds_divider(self) -> i64 {
+    /// Panics if called on a non-time/day field.
+    pub fn micros_multiplier(self) -> i64 {
```

Comment:
Seems like this wants to either be powered by or replaced by `convert_date_time_unit`

Reply:
Good point, I've changed the implementation to use `convert_date_time_unit`